### ISSUE TITLE
Speed up Get_RemoteCheckSum

### ISSUE THREAD

**Issue by chrrobi-sagard** [chocolatey-community/chocolatey-au#58]
### Checklist

- [X] I have verified this is the correct repository for opening this issue.
- [X] I have verified no other issues exist related to my request.

### Is Your Feature Request Related To A Problem? Please describe.

When running Get-RemoteChecksum in the au_beforeupdate() function on a large file, the process takes a long time. In benchmarks undertaken by external pers, it is suggested that using NET's webclient vs Invoke-WebRequest is significantly faster for downloading. For example, for a 150mb file it takes minutes with Invoke_WebRequest vs seconds with the webclient.

### Describe The Solution. Why is it needed?

I would suggest we replace the Invoke-WebRequest call with the .NET webclient call to significantly speed up downloads. This should speed up the checksumming process as part of au_beforeupdate(). This could also be added to other locations which download files if required.

**Comment by TheCakeIsNaOH** [comment]
> I would suggest we replace the Invoke-WebRequest call with the .NET webclient call to significantly speed up downloads.

I don't think this would be a problem, but I am unsure if there would be any behavior changes between these two options that would cause downloads to fail using the webclient (thus making this a breaking change). If I understand correctly, `Invoke-WebRequest` uses the Internet Exporer engine to make the request, while the .NET webclient uses the .NET code to make the request, and given the possibility of buggy download sites, I don't know if both will always work exactly the same. 

Another option may be to disable the progress bar via `$ProgressPreference = 'silentlycontinue'`, which should provide an increase in speed for large files, similar to using the webclient.

**Comment by chrrobi-sagard** [comment]
In looking at results from google, I can see older issues with very large files (gb+) on spotty internet connections. I have not seen recent reports of issues.

I will try without the progress bar to see if there is a speed improvement.

Another option would be to add a flag which uses .NET client if true else uses Invoke-WebRequest? Would allow for testing across some of the automated updates with just a flag change to see at scale if issues within the appveyor builds?

The packages I have tested against (not many) have not caused issues with local runs.
